app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdScreen.kt

```diff
@@ -42,90 +57,65 @@ class ScanSimpleIdScreen(context: Context, attrs: AttributeSet) : ConstraintLayo
   @Inject
   lateinit var effectHandlerFactory: ScanSimpleIdEffectHandler.Factory
 
-  @Inject
-  lateinit var activity: AppCompatActivity
-
   @Inject
   lateinit var scanSimpleIdUpdate: ScanSimpleIdUpdate
 
-  private var binding: ScreenScanSimpleBinding? = null
+  @Inject
+  lateinit var router: Router
 
   private val toolBar
-    get() = binding!!.toolBar
+    get() = binding.toolBar
 
   private val qrCodeScannerViewContainer
-    get() = binding!!.qrCodeScannerViewContainer
+    get() = binding.qrCodeScannerViewContainer
 
   private val shortCodeText
-    get() = binding!!.shortCodeText
+    get() = binding.shortCodeText
 
   private val shortCodeErrorText
-    get() = binding!!.shortCodeErrorText
+    get() = binding.shortCodeErrorText
 
   private val searchingContainer
-    get() = binding!!.searchingContainer
+    get() = binding.searchingContainer
 
   private val keyboardVisibilityDetector = KeyboardVisibilityDetector()
 
   private lateinit var qrCodeScannerView: IQrCodeScannerView
 
-  var scanResultsReceiver: ScanResultsReceiver? = null
+  override fun defaultModel() = ScanSimpleIdModel.create()
 
-  private val events by unsafeLazy {
-    Observable
-        .mergeArray(qrScans(), keyboardEvents(), qrCodeChanges(), doneClicks())
-        .compose(ReportAnalyticsEvents())
-  }
-
-  private val delegate by unsafeLazy {
-    val uiRenderer = ScanSimpleIdUiRenderer(this)
+  override fun uiRenderer() = ScanSimpleIdUiRenderer(this)
 
-    MobiusDelegate.forView(
-        events = events.ofType(),
-        defaultModel = ScanSimpleIdModel.create(),
-        update = scanSimpleIdUpdate,
-        effectHandler = effectHandlerFactory.create(this).build(),
-        modelUpdateListener = uiRenderer::render
-    )
-  }
+  override fun bindView(layoutInflater: LayoutInflater, container: ViewGroup?) =
+      ScreenScanSimpleBinding.inflate(layoutInflater, container, false)
 
-  override fun onAttachedToWindow() {
-    super.onAttachedToWindow()
-    delegate.start()
-  }
+  override fun events() = Observable
+      .mergeArray(qrScans(), keyboardEvents(), qrCodeChanges(), doneClicks())
+      .compose(ReportAnalyticsEvents())
```

Comment:
Should we consider moving the reports of analytics events into the base screen? 🤔

Reply:
Yeah, I was considering doing that later once it's all migrated.